src/client/theme-default/composables/outline.ts

```diff
@@ -30,24 +31,67 @@ export function useOutline() {
   }
 }
 
-export function resolveHeaders(headers: Header[]) {
-  return mapHeaders(groupHeaders(headers))
+export function resolveHeaders(
+  headers: Header[],
+  levelsRange: PageData['frontmatter']['outline'] = 2
+) {
+  const levels: [number, number] =
+    typeof levelsRange === 'number'
+      ? [levelsRange, levelsRange]
+      : levelsRange === 'deep'
+      ? [1, 6]
```

Comment:
Changing `deep` is actually a breaking change for some projects (like vitest) although it wasn't documented. Also if we want to include all headers in `deep`, then I think we should make it `[2, 6]`. Only one `h1` heading per page is usually recommended. It doesn't make sense adding the page title to outline.

Reply:
Add and document both `deep` and `all`? 😅

Reply:
I don't prefer to have both `'deep'` and `'all'` together cause it's really confusing for me 😅. Yeah, let's make `deep` to `[2, 6]` and mark it in the changelog so the Vitest doc maintainer can know it.